docker/Dockerfile

```diff
@@ -93,30 +92,37 @@ RUN OPENSSL_VER=1.1.1g \
  && make test \
  && make install -j$THREADS
 
+RUN if [ "$DISTRO" = "bionic" ] ; then \
+      mkdir /usr/local/lib64; \
+    fi
 
 # -------------=== common/cpp build ===-------------
 
 #Build common/cpp intermediate docker image
-FROM ubuntu:bionic as common_cpp_image
-
-RUN apt-get update \
- && apt-get install -y -q \
-    pkg-config \
-    cmake \
-    make
+FROM $IMAGE as common_cpp_image
+ARG DISTRO
+COPY ./scripts/install_packages /project/avalon/scripts/
 
+RUN packages="pkg-config make cmake "; \
+    if [ "$DISTRO" = "centos" ] ; then \
+      packages="$packages gcc gcc-c++"; \
+    fi; \
+    /project/avalon/scripts/install_packages -c install -q "$packages"
 
 # Copy openssl build artifacts from openssl_image
 COPY --from=openssl_image /usr/local/ssl /usr/local/ssl
 COPY --from=openssl_image /usr/local/bin /usr/local/bin
 COPY --from=openssl_image /usr/local/include /usr/local/include
 COPY --from=openssl_image /usr/local/lib /usr/local/lib
+COPY --from=openssl_image /usr/local/lib64 /usr/local/lib64
 
 RUN ldconfig \
  && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/
 
 ENV TCF_HOME=/project/avalon
 
+ENV PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib64/pkgconfig/
```

Comment:
If `/usr/local/lib64/pkgconfig/`  is specific to centos please put this inside a condition

Reply:
I think "if-else" is not supported inside ENV command.